### DISCUSSION TITLE
Which plugins are stable and still working in the Netbox 3.1

### DISCUSSION THREAD

**Discussion by igorcardoso14** [netbox-community/netbox#8068] [Q&A]
Hi everyone. How are you?

So, which main plugins are working in the new version of the Netbox (3.0/3.1)? 

I using the topology views and the secret plugin. I could some feedback about that two plugins. 

Because receveid some informations about the instability in the plugins on the new flashbox;

Thanks for all.

**Comment by DanSheps** [comment]
Secrets should work, just be sure to load both netbox_plugin_extensions and the secret plugin.

**Reply by greinick** [reply]
Hi, I did the update from 3.0 to 3.1 like i did in the past for all updates. But when accessing the secrets menu i do get an error page:

Server Error:

<class 'django.template.exceptions.TemplateDoesNotExist'>

netbox_plugin_extensions/generic/object_list.html

Python version: 3.8.10
NetBox version: 3.1.0

May be this time I missed a step?

Thanks for some suggestions . Regards

**Reply by DanSheps** [reply]
You need to also load the netbox_plugin_extensions plugin

**Reply by greinick** [reply]
Thanks, may be you could give me a hint how to check if they are enabled/loaded? And may be how to enable them if there has something changed from 3.0 to 3.1?

Currently I have only netbox_secretstore with "_" in local_requirements.txt and configuration.py.

Or could/should I "just rerun" the steps here? https://github.com/dansheps/netbox-secretstore/

Thanks!

**Reply by DanSheps** [reply]
netbox_plugin_extensions needs to be in the configuration.py as well

**Reply by greinick** [reply]
Thanks! I'm bit confused by the  underscore "_" for configuration.py and hyphen "-"  for local_requirements.txt.

Thats correct to have both typings?

**Reply by DanSheps** [reply]
That is correct.

**Reply by greinick** [reply]
O.K. I did edit the two files and as there was/is an update for netbox pending, I installed that and with the modified config and requirenment file.

I did run the netbox update script and reloaded/restarted the deamons. But I still get the error:

<class 'django.template.exceptions.TemplateDoesNotExist'>

inc/table_controls.html

Python version: 3.8.10
NetBox version: 3.1.2


Your help would be much appreciated

**Reply by DanSheps** [reply]
This might be a change brought about by 3.1.2, I will have to take a look

**Reply by YleDebarle** [reply]
I stumbled on this issue while trying to update to 3.1.2 and my plugin was failing the same way.

This template seems to have been deleted, there's only a "table_controls_htmx.html" available there. Using it instead in my template fixed the issue for me at least without any noticeable differences so far.

Not sure if this change is intentional though ?

**Reply by candlerb** [reply]
`inc/table_controls.html` was removed in [this commit](https://github.com/netbox-community/netbox/commit/8ed79d5973c77b489b27fe01ac70cd87ffeacbd5), which is between v3.1.1 and v3.1.2:

```
commit 8ed79d5973c77b489b27fe01ac70cd87ffeacbd5
Author: jeremystretch <jstretch@ns1.com>
Date:   Tue Dec 14 16:44:03 2021 -0500

    Remove obsolete templates
```

**Reply by greinick** [reply]
> This might be a change brought about by 3.1.2, I will have to take a look

Hi, did you had any time yet, to look into that issue? How may I help?

**Comment by kort3x** [comment]
both mentioned plugins are not working for me
